### DISCUSSION TITLE
Yuvarla Gitsin 1759D

### DISCUSSION THREAD

**Discussion by dsudeee** [sonsuzus/sonsuzus#502] [Soru Cevap] [answered]
Berlandia'da enflasyon nedeniyle mağazalar, malların fiyatlarını arttırmak zorundadır. Verilen eski fiyat n'yi, 1 ile m arasında bir tam sayı olan k ile çarparak mümkün olan en yüksek sonunda sıfır sayısına sahip yeni fiyatı bulmak amaçlanmaktadır. Eğer daha "yuvarlak" bir fiyat elde etmek imkânı yoksa, en yüksek fiyat n⋅m olarak döndürülmelidir.

Ana Noktalar
Enflasyondan etkilenen Berlandia'da mağazalar fiyatlarını ayarlamak zorundadır.
Eski fiyat n, belirli bir k ile çarpılarak yeni fiyatın sonunda maksimum sıfır sayısına sahip olması hedeflenmektedir.

1 ≤ k ≤ m koşuluna bağlı olarak k'nın en küçük olması gerekmektedir; bu da sıfırların sayısını arttıracaktır.

Eğer daha yuvarlak bir fiyat elde etmek mümkün değilse, n⋅m (maksimum fiyat) sonuç olarak verilecektir.

Her bir test durumu, eski fiyat n ve maksimum çarpan m değerlerini içermektedir.
Verilen örneklerde, farklı n ve m kombinasyonları için en iyi yeni fiyatlar gösterilmiştir.

Tüm testlerin çıktıları, her bir durumda hesaplanan yeni fiyatları ayrı satırlarda göstermektedir.

```
Girdi
10
6 11
5 43
13 5
4 16
10050 12345
2 6
4 30
25 10
2 81
1 7
```

```
Çıktı
60
200
65
60
120600000
10
100
200
100
7
```

**Comment by dsudeee** [comment, marked as answer]
```
n = int(input())
for i in range(n):
    a, b = map(int, input().split())
    sifir = 0
    maxsifir = 0
    for j in range(b,1,-1):
      text = str(a*j)
      say = -1
      sifir = 0
      while text[say] == '0':
        sifir += 1
        say = say - 1
      if sifir > maxsifir:
        maxsifir = sifir
        sonuc = a*j
    if maxsifir == 0:
      sonuc = a*b
    print(sonuc)
```

**Reply by sonsuzus** [reply]
Harika.
